packages/vite-plugin/src/proxy-middleware.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * This modules provides the proxy located at
+ * /motion-canvas-proxy/...
+ *
+ * It is needed when acessing remote resources.
+ * Trying to access remote resources works while
+ * in preview, but will fail when you try to
+ * output the image (= "read" the canvas)
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image
+ * for reasons
+ *
+ * Using the proxy circumvents CORS-issues because
+ * this way all remote resources are served from the
+ * same host as the main app.
+ */
+
+import {Connect} from 'vite';
+import axios from 'axios';
+import {ServerResponse, IncomingMessage} from 'http';
+
+/**
+ * Configuration used by the Proxy plugin
+ */
+export interface MotionCanvasCorsProxyPluginOptions {
```

Comment:
It should be called `MotionCanvasCorsProxyOptions` since it's not technically a standalone plugin. It's part of the main Vite plugin.